internal/datasource/os/files.go

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package os
+
+import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"path"
+
+	"github.com/nginx/agent/v3/api/grpc/instances"
+)
+
+// map of files with filepath as key
+type FileCache = map[string]*instances.File
```

Comment:
should this type be visible outside of this file?

Reply:
Yes as its needed in config writer